handlers/load_test_preferences_handler.go

```diff
@@ -75,3 +80,197 @@ func (h *Handler) LoadTestPrefencesHandler(w http.ResponseWriter, req *http.Requ
 
 	_, _ = w.Write([]byte("{}"))
 }
+
+type custTestConf struct {
+	Val *SMPS.PerformanceTestConfig
+}
+
+func (c *custTestConf) MarshalJSON() ([]byte, error) {
+	m := jsonpb.Marshaler{
+		EmitDefaults: true,
+	}
+	val, err := m.MarshalToString(c.Val)
+	return []byte(val), err
+}
+
+// UserTestPreferenceHandler is used for persisting load test preferences
+func (h *Handler) UserTestPreferenceHandler(w http.ResponseWriter, req *http.Request, prefObj *models.Preference, user *models.User, provider models.Provider) {
+
+	if req.Method == http.MethodPost {
+
+		q := req.FormValue("qps")
+		qps, err := strconv.Atoi(q)
+		if err != nil {
+			err = errors.Wrap(err, "unable to parse qps")
+			logrus.Error(err)
+			http.Error(w, "please provide a valid value for qps", http.StatusBadRequest)
+			return
+		}
+		if qps < 0 {
+			http.Error(w, "please provide a valid value for qps", http.StatusBadRequest)
+			return
+		}
+
+		dur := req.FormValue("t")
+		durT, err := time.ParseDuration(dur)
+		if err != nil {
+			err = errors.Wrap(err, "unable to parse t as a duration")
+			logrus.Error(err)
+			http.Error(w, "please provide a valid value for t", http.StatusBadRequest)
+			return
+		}
+
+		endpoint := req.FormValue("endpoint")
+		if endpoint == "" {
+			logrus.Error("unable to find endpoint")
+			http.Error(w, "please provide a value for endpoint", http.StatusBadRequest)
+			return
+		}
+
+		cu := req.FormValue("c")
+		c, err := strconv.Atoi(cu)
+		if err != nil {
+			err = errors.Wrap(err, "unable to parse c")
+			logrus.Error(err)
+			http.Error(w, "please provide a valid value for c", http.StatusBadRequest)
+			return
+		}
+		if c < 0 {
+			http.Error(w, "please provide a valid value for c", http.StatusBadRequest)
+			return
+		}
+
+		gen := req.FormValue("gen")
+		// TODO: after we have interfaces for load generators in place, we need to make a generic check, for now using a hard coded one
+		var loadGen SMPS.PerformanceTestConfig_Client_LoadGenerator
+		switch gen {
+		case "fortio":
+			loadGen = SMPS.PerformanceTestConfig_Client_fortio
+		case "wrk2":
+			loadGen = SMPS.PerformanceTestConfig_Client_wrk2
+		default:
+			logrus.Error("invalid value for gen")
+			http.Error(w, "please provide a valid value for gen (load generator)", http.StatusBadRequest)
+			return
+		}
+
+		proto := req.FormValue("protocol")
+		var protocol SMPS.PerformanceTestConfig_Client_Protocol
+		switch proto {
+		case "HTTP":
+			protocol = SMPS.PerformanceTestConfig_Client_http
+		case "gRPC":
+			protocol = SMPS.PerformanceTestConfig_Client_grpc
+		case "TCP":
+			protocol = SMPS.PerformanceTestConfig_Client_tcp
+		default:
+			logrus.Error("invalid value for protocol")
+			http.Error(w, "please provide a valid value for gen (load generator)", http.StatusBadRequest)
```

Comment:
Wrong error message over here

Reply:
Instead of reading in these values as form input, I am considering to read it as JSON, what do you suggest?